QuEST/src/QuEST.c

```diff
@@ -834,6 +834,18 @@ Complex calcInnerProduct(Qureg bra, Qureg ket) {
     return statevec_calcInnerProduct(bra, ket);
 }
 
+qreal calcDensityInnerProduct(Qureg bra, Qureg ket) {
+    validateMatchingQuregTypes(bra, ket,  __func__);
```

Comment:
Since this function has ***Density*** in the title, I think it should only accept density matrices. 
I.e. this line should become two:
```C
validateDensityMatrQureg(bra, __func__);
validateDensityMatrQureg(ket, __func__);
```
It's not even performing the inner product when given state-vectors, really!

Reply:
Addressed this in acf7f95214544204939c8267381218a72ec98096.